### DISCUSSION TITLE
How to Stop an Animation

### DISCUSSION THREAD

**Discussion by j4587698** [AvaloniaUI/Avalonia#17038] [Q&A]
I created an animation like this:
```
<Style Selector="Image#AlbumCover">
  <Style.Animations>
    <Animation IterationCount="Infinite" Duration="0:0:40">
      <KeyFrame Cue="0%">
        <Setter Property="RotateTransform.Angle" Value="0" />
      </KeyFrame>
      <KeyFrame Cue="100%">
        <Setter Property="RotateTransform.Angle" Value="360" />
      </KeyFrame>
    </Animation>
  </Style.Animations>
</Style>
```
How can I control the start and stop of the animation using code?

I looked into it, and it seems that if I switch to creating the animation with code, I can use RunAsync to start the animation. However, there doesn't appear to be a stop method. How should I stop the animation?

**Comment by maxkatz6** [comment]
`RunAsync` method accepts cancellation token.
You also can stop animation via styles by applying/removing style-class.
